src/server/minecraft-server.js

```diff
@@ -0,0 +1,1503 @@
+                    if (eulaUrlLine.length == 2) {
+                        properties.eulaUrl = 'https://' + eulaUrlLine[1].substr(0, eulaUrlLine[1].indexOf(')'));
+                        await this.log('MinecraftServer EULA location: ' + properties.eulaUrl);
+                    }
+                    line = eula[lineNumber].split('=');
+                    if (line.length == 2) {
+                        properties.acceptedEula = !!JSON.parse(line[1]);
+                        await this.log(`MinecraftServer EULA accepted? ${properties.acceptedEula}`);
+                    }
+                }
+            }
+            properties.eulaFound = true;
+            properties.installed = true;
+            return Promise.resolve();
+        } catch (err) {
+            await this.log('Failed to read eula.txt.');
+            await this.log(err);
+            await this.log('Minecraft probably needs to be run once to stage new files.');
+            await this.log('Use the web interface to start the Minecraft server and accept the license agreement.');
+            properties.eulaFound = false;
+            properties.installed = false;
+            return Promise.reject(err);
+        }
+    }
+
+    async getMinecraftVersions () {
+        // TODO enable snapshot updates with a property/preference
+        let minecraftVersions = [];
+        let releaseVersions = [];
+        let snapshotVersions = [];
+
+        try {
+            await this.log('Getting releases available from Minecraft...');
+            let response = await axios.get('https://launchermeta.mojang.com/mc/game/version_manifest.json');
+            minecraftVersions = response.data;
+            minecraftVersions.versions.forEach(version => {
+                if (version.type === 'release') {
+                    releaseVersions.push(version);
+                } else if (version.type === 'snapshot') {
+                    snapshotVersions.push(version);
+                }
+            });
+            this.properties.versions = {
+                installed: this.properties.versions.installed || null,
+                latest: minecraftVersions.latest,
+                release: releaseVersions,
+                snapshot: snapshotVersions };
+            await this.log('Got releases from Minecraft.');
+            return Promise.resolve(this.properties.versions);
+        } catch (err) {
+            await this.log(`Unable to get release information from https://launchermeta.mojang.com/mc/game/version_manifest.json`);
+            return Promise.reject(err);
+        }
+    }
+
+    async getOps () {
+        await this.log('Getting MinecraftServer ops...');
+
+        let properties = this.properties;
+        let minecraftDirectory = properties.settings.minecraftDirectory;
+
+        if (properties.installed) {
+            try {
+                properties.ops = JSON.parse(await fs.readFile(path.join(minecraftDirectory, 'ops.json'), 'utf8'));
```

Comment:
Ditto about the `readJson` functions. There's probably more of these, but I can't mark them all right now